libraries/AP_Math/spline5.cpp

```diff
@@ -0,0 +1,61 @@
+/*
+This work by Ryan Muller released under the Creative Commons CC0 License
+http://creativecommons.org/publicdomain/zero/1.0/
+*/
+
+/**
+  Spline interpolation of a parametric function.
+
+  INPUT: std::vector<double> x
+        A list of double values that represent sampled
+        points. The array index of each point will be
+        taken as the parameter t by which x will be
+        represented as a function.
+
+  OUTPUT: std::vector<cv::Vec4d> P
+        A list of cv::Vec4d representing polynomials. To
+        interpret segment [i]:
+        x(t) =  P0*a + P1*b + P2*(a^3-a)/6 + P3*(b^3-b)/6
+        where a = t-i
+              b = i-t+1
+  */
+
+#include "spline5.h"
+
+void splinterp5(const float x[5], float out[5][4]){
+
+    // loop counter
+    int i;
```

Comment:
we try to avoid 'int' in our code as it varies in size between platforms. Instead use uint8_t, uint16_t etc.